src/renderers/dom/client/ReactMount.js

```diff
@@ -724,6 +729,15 @@ var ReactMount = {
       }
     }
   },
+  /**
+   * Rather than count mounts and unmounts, we query the dom for the DOMProperty.ROOT_ATTRIBUTE_NAME. This is used to
+   * dispose of global event listeners, when the last component is removed/unmounted.
+   * @returns Returns the number of react tagged root elements.
+   * @private
+   */
+  _countReactRootElements: function() {
+    return document.querySelectorAll('[ROOT_ATTR_NAME]').length;
+  },
```

Comment:
> IE8 does not have it, 

Does react support ie8, must i provide a fallback ?

> Wouldn't it make more sense to just keep a counter?
The problem i was trying to avoid by counting is that only works if code unmounts each and every component helping the counter stay accurate. Im not sure how big or small a concern this is. For me it seems wrong to track and "count" when you can ask the dom to tell you and that would never be wrong.

What should i do ?

1) update code to "count" root additions and removals ?
2) keep doc.querySelectAll and provide a fallback for ie ?